### PR TITLE
Create changelog.txt

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
As promised a draft of the changelog that we have been discussing. 

I'm sure I've forgotten a few things and that there are improvements to be done, especially in the changed/added
section, so please don't hesitate to have a look :)

### CODE REVIEW

**Review by rhfogh — APPROVED** [review]
Looks good

File: changelog.txt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+MXCuBE is a project with a long history, initially started at ESRF in 2003 and built
+on top of a inhouse developed framework called Framework-2. The project has since
+been adopted by a large number of sites performing MX experiments. There is within
+MXCuBE the possibility to create encapsulated pieces of code that pertain to certain
+equipment or routines. This logic is written as what's called HardwareObjects. These
+HardwareObjects can be configured via XML (and more recently YAML) files, there is
+further an infrastructure for emitting events from and to HardwareObjects. Each site
+has the possibility to extend and adapt the functionality by adding site specific
+HardwareObects.
```
Comment (meghdadyazdi) [inline changelog.txt:17]: HardwareOjbects

File: changelog.txt [hunk 2]
```diff
@@ -0,0 +1,121 @@
+CHANGELOG (DRAFT)
+
+===========
+
+Background
+
+-----------------
+
+MXCuBE is a project with a long history, initially started at ESRF in 2003 and built
+on top of a inhouse developed framework called Framework-2. The project has since
```
Comment (meghdadyazdi) [inline changelog.txt:10]: an inhouse

File: changelog.txt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+MXCuBE the possibility to create encapsulated pieces of code that pertain to certain
+equipment or routines. This logic is written as what's called HardwareObjects. These
+HardwareObjects can be configured via XML (and more recently YAML) files, there is
+further an infrastructure for emitting events from and to HardwareObjects. Each site
+has the possibility to extend and adapt the functionality by adding site specific
+HardwareObects.
+
+During the course of time parts of the API have been "organically" patched and grown
+to be less consistent. The MXCuBE developers team therefore decided that it would be
+crucial to refactor and cleanup up inconsistencies and rework other aspects of the
```
Comment (meghdadyazdi) [inline changelog.txt:21]: and clean up

File: changelog.txt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+core functionality (known as HardwareRepository) to facilitate the  further
```
Comment (meghdadyazdi) [inline changelog.txt:22]: to facilitate further

File: changelog.txt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+The MXCuBE developers conducted a series of meetings and workshops to discuss these
+matters before starting the initial work. The discussions have naturally continued
+during the process and some are still ongoing. This means that there are still
+features and ideas that are being discussed that are not yet settled but are likely
+to be implemented soon, found in the the BEING DRAFTED section. Finally, the aim of
+this rework is to make it easier and faster to extend MXCuBE with new functionality.
+It will also contribute to make it easier for new sites to adopt MXCuBE.
+
+During this process a more rigorous versioning schema have been put in place to
```
Comment (meghdadyazdi) [inline changelog.txt:33]: schema has been

File: changelog.txt [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+facilitate the update and maintenance process of the software on the various sites.
+This file is the first changelog and are meant to summarise the changes made in a
```
Comment (meghdadyazdi) [inline changelog.txt:35]: is meant

File: changelog.txt [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+comprehensible way so that any developer can get an idea of what has changed, being
+discussed and are still subject to change. It's of course impossible to summarize
```
Comment (meghdadyazdi) [inline changelog.txt:37]: It's, of course, impossible

File: changelog.txt [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+all the work that have been done during the course of the last 2 years but the
```
Comment (meghdadyazdi) [inline changelog.txt:38]: all the work that has been done

File: changelog.txt [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+- HardwareRepository was renamed to mxcubecore and made into a Python module (installable with i.e pip)
+
+- Possibility to configure HardwareObjects using YAML
+
+- All method names and variables are snake cased
+
+- Methods that set a value of a member are prefixed with set_[member name] and members
+    that retrieve a value are called get_[member name].
+
+- The HardwareObject base class have been refactored so that all hardware objects have
```
Comment (meghdadyazdi) [inline changelog.txt:56]: has been refactored

File: changelog.txt [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+      update_state(state=None)
+      update_specific_state(state=None)
+      re_emit_values()
+      force_emit_signals()
+      emit(signal, *args)
+      connect(sender, signal, slot=None)
+      disconnect(sender, signal, slot=None)
+
+   The AbstractActuator base class implements the functionality required by hardware objects
+   that acts as actuators or more specifically have a value. This includes a value, its
```
Comment (meghdadyazdi) [inline changelog.txt:81]: that act as actuators

**Pull request merged**